lib/runtime/datatip.js

```diff
@@ -0,0 +1,93 @@
+/** @babel */
+
+import { Range, Point } from 'atom'
+
+import { client } from '../connection'
+import modules from './modules'
+import { scopes, words } from '../misc'
+
+const datatip = client.import('datatip')
+const { isValidWord } = words
+
+// @FIXME: This does not work with non-ascii characters
+const wordRegex = /[^0-9A-Za-z_!\.´@]/
+const grammar = atom.grammars.grammarForScopeName('source.julia');
+
+/**
+ * Returns `true` if the scope at `bufferPosition` in `editor` is valid code scope to be inspected
+ */
+function isValidScope (editor, bufferPosition) {
+  const scps = editor
+    .scopeDescriptorForBufferPosition(bufferPosition)
+    .getScopesArray();
+  return scopes.isValidInspectionScope(scps)
+}
+
+class DatatipProvider {
+  providerName = 'julia-client-datatip-provider';
+
+  priority = 100;
+
+  grammarScopes = atom.config.get('julia-client.juliaSyntaxScopes');
+
+  useAtomIDEUI = false;
+
+  async datatip(editor, bufferPosition, mouseEvent) {
+    // If Julia is not running, do nothing
+    if (!client.isActive()) return
+
+    // If the scope at `bufferPosition` is not valid code scope, do nothing
+    if (!isValidScope(editor, bufferPosition)) return
+
+    // Code to be inspected
+    const { row, column } = bufferPosition
+    const text = editor.lineTextForBufferRow(row)
+    if (!text) return null
+    let start = column
+    const identifierStart = text.slice(0, column).split('').reverse().join('').search(wordRegex)
+    if (identifierStart === -1) {
+      start = 0
+    } else {
+      start -= identifierStart
+    }
+    let end = column
+    const identifierEnd = text.slice(column).search(wordRegex)
+    if (identifierEnd === -1) {
+      end = text.length
+    } else {
+      end += identifierEnd
+    }
+    const word = text.slice(start, end)
```

Comment:
It would probably be easiest to do something like [this](https://github.com/atom/atom/blob/v1.38.2/src/cursor.js#L606-L616) with the regex already defined in words.js.

Reply:
With [this commit](https://github.com/JunoLab/atom-julia-client/pull/573/commits/94641a17415b741f382f87c968e8584295c8e3cc), the regex and index stuff turn to work with non-ascii characters.
![Screenshot 2019-07-22 16 18 09](https://user-images.githubusercontent.com/40514306/61613686-f4982c80-ac9c-11e9-8cd2-790a717000c1.png)

But yes, it would be better to borrow the regex from word.js.
I will change the implementation if I have time.

Reply:
Yeah, solved in [this commit](https://github.com/JunoLab/atom-julia-client/pull/573/commits/2f938bf31b6f91105cb8fdd0d3d5386aa8a5256d).
Thanks for the suggestion.

Reply:
![image](https://user-images.githubusercontent.com/40514306/61708943-08b95800-ad89-11e9-8550-b88034166d70.png)